src/de/mod.rs

```diff
@@ -819,49 +935,69 @@ mod tests {
         assert_eq!(crate::from_str(r#" "thing" "#), Ok((Type::Thing, 9)));
     }
 
+    #[test]
+    fn char() {
+        assert_eq!(crate::from_str("\"n\""), Ok(('n', 3)));
+        assert_eq!(crate::from_str("\"\\\"\""), Ok(('"', 4)));
+        assert_eq!(crate::from_str("\"\\\\\""), Ok(('\\', 4)));
+        assert_eq!(crate::from_str("\"/\""), Ok(('/', 3)));
+        assert_eq!(crate::from_str("\"\\b\""), Ok(('\x08', 4)));
+        assert_eq!(crate::from_str("\"\\f\""), Ok(('\x0C', 4)));
+        assert_eq!(crate::from_str("\"\\n\""), Ok(('\n', 4)));
+        assert_eq!(crate::from_str("\"\\r\""), Ok(('\r', 4)));
+        assert_eq!(crate::from_str("\"\\t\""), Ok(('\t', 4)));
+        assert_eq!(crate::from_str("\"\\u000b\""), Ok(('\x0B', 8)));
+        assert_eq!(crate::from_str("\"\\u000B\""), Ok(('\x0B', 8)));
+        assert_eq!(crate::from_str("\"\u{3A3}\""), Ok(('\u{3A3}', 4)));
```

Comment:
Can you use the `r#""#` syntax for these tests so its more clear what's supposed to be passed to the JSON decoder? I'm struggling to see what's supposed to be there and what's supposed to be escaped.